fix(export): show 'Data export requires Business Plan' on 404 for SaaS

JoshFerge opened the pull request:
Right now it is confusing that a user can try and export tags but we do not tell them why. if we're on SaaS, send a more helpful error message.

Ideally, we should probably just gray out the button entirely if the user is not on business with a tooltip.

JoshFerge committed:
dataExport: show 'Data export requires Business Plan' on 404 for SaaS; add test; keep generic/self-hosted fallback

scttcper commented on static/app/components/dataExport.tsx:
```diff
@@ -70,11 +71,14 @@ export function useDataExport({
-            "We tried our hardest, but we couldn't export your data. Give it another go."
-          );
+          err?.status === 404 && !isSelfHosted
+            ? t('Data export requires Business Plan')
```
shouldn't the button be disabled or removed if they can't export on their plan

JoshFerge replied:
yeah that is probably the better fix

JoshFerge replied:
i just updated the branch. let me know if that makes sense.

JoshFerge committed:
dataExport: gate export button with Feature and disable on missing discover-query; show GS hovercard via hookName

JoshFerge committed:
dataExport: revert to not rendering anything for unauthorized orgs (no hovercard); keep button enabled when feature present

roggenkemper reviewed (commented):
i think the only change we need to make here is actually disabling/hiding `TagExportDropdown` if they don't have the feature flag. the button in discover is probably gated, but it's not in the tag drawer on issue details. with that, we shouldn't even need to update the error message since they won't be able to attempt to export

JoshFerge committed:
fix

JoshFerge committed:
tagExportDropdown: hide when org lacks organizations:discover-query; add spec. dataExport: remove 404 Business Plan messaging and test; rely on gating

roggenkemper commented on static/app/views/issueDetails/groupDistributions/tagExportDropdown.tsx:
```diff
@@ -30,39 +31,41 @@ export default function TagExportDropdown({tagKey, group, organization, project}
-              `/${organization.slug}/${project.slug}/issues/${group.id}/tags/${tagKey}/export/`,
-              '_blank'
-            );
+    <Feature features="organizations:discover-query" organization={organization}>
```
`Feature` is already wrapped with `withOrganization` so we shouldn't have to pass organization in here

JoshFerge replied:
okay i believe i updated it. but this will just hide the button right. do we have an example of showing an "upgrade to business" hook?

roggenkemper replied:
i don't think we have a unified upgrade to business hook since each place that I know of that has this upsell, does it slightly differently

JoshFerge committed:
fix: Remove redundant organization prop from Feature component

The Feature component is already wrapped with withOrganization HOC,
so passing organization prop explicitly is redundant.

Update tests to provide organization context through render options.
